### DISCUSSION TITLE
Where to put field documentation

### DISCUSSION THREAD

**Discussion by danieldaeschle** [vlang/v#7245] [General]
Short field docs are behind the field.
```v
struct Bar {
    foo string // foo does something.
}
```

But what about long comments?
Last discussion with Alex was about moving it above if the comment is too long.
```v

struct Bar {
    foo string // foo does something.
    // milk is different from foo because the comment is too long to put it behind the field. That's why it's on above the field.
    milk Milk
}
```

Does vdoc already handle this?

**Comment by larpon** [comment]
Long comments could also be split around first available white space at some length:
```v
struct Bar {
    foo string // foo does something.
    // milk is different from foo because the comment is too long
    // so `vfmt` could break it at some length to put it behind the field.
    // That's why it's above the field.
    milk Milk
}
```

**Reply by larpon** [reply]
This would OFC not work with `// extremely_long_connected_sentence_that_some_programmer_will_surely_use_to_get_around_it`

**Comment by spytheman** [comment]
My first instinct while reading the above 'milk' comment is as a continuation of the previous line, just like lines in a book.
The commented area seem to me like one large block, not like 2 independent blocks, belonging first to the foo line, and the other to the milk line.

**Reply by larpon** [reply]
Oh yeah. Then we'd maybe need a newline between.

```v
struct Bar {
    foo string // foo does something.

    // milk is different from foo because the comment is too long
    // so `vfmt` could break it at some length to put it behind the field.
    // That's why it's above the field.
    milk Milk
}
```

**Comment by spytheman** [comment]
btw, are there natural written languages, where reading direction changes in the middle of the page?

**Reply by danieldaeschle** [reply]
> btw, are there natural written languages, where reading direction changes in the middle of the page?

😂 don‘t know

Nice opinion. What is your suggestion then for longer comments.

**Reply by danieldaeschle** [reply]
Btw. Most languages have doc string above. What about doing it always?

**Reply by larpon** [reply]
I don't think there is but now that it's on the internet - someone will probably go ahead and invent one just to keep us programmers on our toes 😅

**Reply by larpon** [reply]
I really like comments after the field BTW

**Reply by RiordanIX** [reply]
Historically, yes. It's called [Boustrophedon](https://en.wikipedia.org/wiki/Boustrophedon). Basically not used in modern day except for some conlangs.
